activerecord/lib/active_record/model_schema.rb

```diff
@@ -435,15 +439,27 @@ def reset_column_information
         initialize_find_by_cache
       end
 
+      protected
+
+        def initialize_load_schema_monitor
+          @load_schema_monitor = Monitor.new
+        end
+
       private
 
+        def inherited(child_class)
+          super
+          child_class.initialize_load_schema_monitor
```

Comment:
Exactly. Separate classes populating their columns at the same time is perfectly safe, so it's better that we allow it -- otherwise we introduce avoidable delays.

(Likewise on the double-checked locking in `load_schema`: once the schema is known to be fully loaded, we don't want to delay anyone by forcing them to acquire a lock just to learn that.)